configure.ac

```diff
@@ -5127,7 +5127,68 @@ if test "$Py_DEBUG" = 'true' -a "$with_trace_refs" != "yes"; then
 fi
 
 AC_SUBST(EXT_SUFFIX)
-EXT_SUFFIX=.${SOABI}${SHLIB_SUFFIX}
+EXT_SUFFIX="."
+# Special case of EXT_SUFFIX with python build with mingw. 
+# Python can with compiled with clang or gcc and linked
+# to msvcrt or ucrt. To avoid conflicts between them
+# we are selecting the extension as based on the compiler
+# and the runtime they link to
+#   gcc + x86_64 + msvcrt = cp{version number}-x86_64
+#   gcc + i686 + msvcrt = cp{version number}-i686
+#   gcc + x86_64 + ucrt = cp{version number}-x86_64-ucrt
+#   clang + x86_64 + ucrt = cp{version number}-x86_64-clang
+#   clang + i686 + ucrt = cp{version number}-i686-clang
+
+case $host in
+  *-*-mingw*)
+    # check if we are linking to ucrt
+    AC_MSG_CHECKING(whether linking to ucrt)
+    AC_RUN_IFELSE([AC_LANG_SOURCE([[
+    #include <stdio.h>
+    int main(){
+      _UCRT;
+    }
+    ]])],[linking_to_ucrt=yes],[linking_to_ucrt=no])
+    AC_MSG_RESULT($linking_to_ucrt)
+    ;;
+esac
+
+AC_MSG_CHECKING(EXT_SUFFIX)
+VERSION_NO_DOTS=$(echo $LDVERSION | tr -d .)
+case $host in
+i686-*-mingw*)
+  if test -n "${cc_is_clang}"; then
+    # it is CLANG32
+    EXT_SUFFIX+="cp${VERSION_NO_DOTS}-i686-clang"
+  else
+    if test $linking_to_ucrt = no; then
+      EXT_SUFFIX+="cp${VERSION_NO_DOTS}-i686"
+    else
+      EXT_SUFFIX+="cp${VERSION_NO_DOTS}-i686-ucrt"
+    fi
+  fi
+  ;;
+x86_64-*-mingw*)
+  if test -n "${cc_is_clang}"; then
+    # it is CLANG64
+    EXT_SUFFIX+="cp${VERSION_NO_DOTS}-x86_64-clang"
+  else
+    if test $linking_to_ucrt = no; then
+      EXT_SUFFIX+="cp${VERSION_NO_DOTS}-x86_64"
+    else
+      EXT_SUFFIX+="cp${VERSION_NO_DOTS}-x86_64-ucrt"
+    fi
+  fi
+  ;;
+aarch64-*-mingw*)
+  EXT_SUFFIX+="cp${VERSION_NO_DOTS}-aarch64"
+  ;;
+*)
+  EXT_SUFFIX+=.${SOABI}
```

Comment:
Shouldn't this all not set `SOABI` instead of `EXT_SUFFIX`?

Reply:
Changing `SOABI` broke building wheels. https://github.com/msys2-contrib/cpython-mingw/pull/3#issuecomment-867422161
Wheel excepts SOABI to be start with `cpython-*` https://github.com/pypa/wheel/blob/master/src/wheel/bdist_wheel.py#L98

Not sure what to do

Reply:
> Should we define `SOABI` first of all? Is it used anywhere? I could see it isn't there in MSVC python.

hard to say. Ideally it would be there for consistency.. but if that doesn't work we can remove it.

Reply:
For now, we can change only the EXT_SUFFIX later I will make a PR to change wheel tag or platform tags where I will see if changing `SOABI` works there.